.github/workflows/main.yml

```diff
@@ -113,6 +113,28 @@ jobs:
             .ci/check-format.sh
       shell: bash
   
+  static-analysis:
+    needs: [detect-code-related-file-changes]
+    if: needs.detect-code-related-file-changes.outputs.has_code_related_changes == 'true'
+    runs-on: ubuntu-22.04
```

Comment:
Specify `ubuntu-latest` to allow the recent clang tools.

Reply:
Done. 

The commit message was also amended to document the reason.